strawberry/schema/execute.py

```diff
@@ -256,3 +259,135 @@ def execute_sync(
         errors=execution_context.result.errors,
         extensions=extensions_runner.get_extensions_results_sync(),
     )
+
+
+async def subscribe(
+    schema: GraphQLSchema,
+    *,
+    extensions: Sequence[Union[Type[SchemaExtension], SchemaExtension]],
+    execution_context: ExecutionContext,
+    process_errors: Callable[[List[GraphQLError], Optional[ExecutionContext]], None],
+) -> AsyncGenerator[Tuple[bool, ExecutionResult], None]:
+    """
+    The graphql-core subscribe function returns either an ExecutionResult or an
+    AsyncGenerator[ExecutionResult, None].  The former is returned in case of an error
+    during parsing or validation.
+    Because we need to maintain execution context, we cannot return an
+    async generator, we must _be_ an async generator.  So we yield a
+    (bool, ExecutionResult) tuple, where the bool indicates whether the result is an
+    potentially multiple execution result or a single result.
+    A False value indicates an single result, most likely an intial
+    failure (and no more values will be yielded) whereas a True value indicates a
+    successful subscription, and more values may be yielded.
+    """
+
+    extensions_runner = SchemaExtensionsRunner(
+        execution_context=execution_context,
+        extensions=list(extensions),
+    )
+
+    # unlike execute(), the entire operation, including the results hooks,
+    # is run within the operation() hook.
+    async with extensions_runner.operation():
+        # Note: In graphql-core the schema would be validated here but in
+        # Strawberry we are validating it at initialisation time instead
+        assert execution_context.query is not None
+
+        async with extensions_runner.parsing():
```

Comment:
This is duplicated from async execution, probably can be reused... see https://github.com/nrbnlulu/strawberry/blob/support_extensions_on_subscriptions/strawberry/schema/execute.py#L192

Reply:
I'll see if I can do that after I change the tuple semantics.

Reply:
I can't see how this code can be reused.  Subscription in my case is an _async generator_ and necessarily _yields_ its result.
Unless, you are willing to make everything use async generator context and create a special single-instance yielder when performing `exeucte()`

Reply:
I did refactor the common validation tests